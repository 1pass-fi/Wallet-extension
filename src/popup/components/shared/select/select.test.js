import '@babel/polyfill'
import { fireEvent, getByText, render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import React from 'react'
import configureStore from 'redux-mock-store'

import { Select } from '.'

describe('Test for ImportPhraseLockScreen component', () => {
  let container, options, defaultOption, label, placeholder, className, onChange
  beforeEach(() => {
    options = ['one', 'two']
    defaultOption = 'one'
    label='label'
    placeholder='placeholder'
    className='className'
    onChange=jest.fn()

    container = render(<Select
      options={options}
      defaultOption={defaultOption}
      label={label}
      placeholder={placeholder}
      className={className}
      onChange={onChange}
    />).container
  })

  describe('renders without crashing', () => {
    it('renders correctly', () => {
      expect(container).toMatchSnapshot()
    })
  })
})
