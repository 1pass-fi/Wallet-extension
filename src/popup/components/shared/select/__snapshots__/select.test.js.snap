// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for ImportPhraseLockScreen component renders without crashing renders correctly 1`] = `
<div>
  <div
    class="select-container className"
  >
    <div
      class="label"
    >
      label
    </div>
    <div
      class="select-wrapper"
    >
      <div
        class="header"
      >
        <input
          class="selected-option"
          placeholder="placeholder"
          readonly=""
          type="text"
          value="one"
        />
        <button
          class="arrow-button"
        >
          <div
            class="arrow-icon"
            style="transform: none;"
          >
            <svg />
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;
